gslib/utils/text_util.py

```diff
@@ -166,7 +164,9 @@ def EncodeStringAsLong(string_to_convert):
   Returns:
     Long that represents the input string.
   """
-  return long(string_to_convert.encode('hex'), 16)
+  if six.PY2:
+    return long(codecs.encode(string_to_convert, 'hex_codec'))
+  return int(codecs.encode(string_to_convert, 'hex_codec'))
```

Comment:
Similar comment to the one above.  I think this might work:

```

hex_bytestr = codecs.encode(six.ensure_binary(string_to_convert), 'hex_codec')
if six.PY3:
  long = int
# Note that `long` accepts either `bytes` or `unicode` as the
# first arg in both py2 and py3:
return long(hex_bytestr, 16)
```

Reply:
Whoops dang, you're right, my logic is a bit doofed! Thanks for being smart and catching that 🙂